documentation/adr/ADR-0004.md

```diff
@@ -0,0 +1,39 @@
+## Title
+
+Converting the repository into an apt-based application (.deb package)
+
+## Proposal status
+
+What is the status of this decision?
+
+-   Proposed
+
+## Implementation status
+
+-   Work in progress
+
+## Context
+
+Our current application utilizes CURL installation methood and the legacy installation method that is cloning the repository and installing each dependency individually. This requires frequent maintenance by the maintainer and necessitates the manual installation of the dependencies. Switching to packaging, specifically to ".deb" packages, would streamline this process by pre-processing all the dependencies and install the application with all the necessary package dependency. This will eliminate the need for manual intervention by the maintainer unless a new dependency was added/contributed on the open-source repository for each system.
+
+## Decision
+
+I propose converting the current CURL installation method to an APT-based installation. APT based installations are more secure as they involve signed packages, while CURL installations directly download from potentially unknown sources. Additionally, APT allows for easy rollback to previous versions in case a newer version is found to be buggy, providing a reliable way to manage software updates and maintain system stability. Moreover, APT-based installations are system-wide, making the software available to all users on the system.
```

Comment:
Questions on this proposal:

- Why are we just looking at a `.deb` package and `apt` as a package manager?
- Given the multiple installation options available, what makes this specifically easier to manage in terms of updates or rolling back versioning? Can't we do that through GitHub?
- Although packages have signing for security, is this necessary right now? If so, why?